components/inbox-mail/InboxMailView.tsx

```diff
@@ -0,0 +1,274 @@
+
+import { useCallback, useEffect, useMemo, useState } from "react";
+import CreateNewMailModal from "./CreateNewMailModal";
+import { InboxMail, InboxMailThread, User, InboxMailThreadSupportProgressState } from "./types-mail";
+import { getInboxMailOverviewByThreadsPaginated, getInboxMailsByThread, createInboxMailByThread, updateInboxMailThreadProgress, deleteInboxMailById } from "./service-mail";
+import KernButton from "../kern-button/KernButton";
+import { MemoIconPlus, MemoIconRefresh, MemoIconHelpCircle } from "../kern-icons/icons";
+import useRefState from "../../hooks/useRefState";
+import { MAIL_LIMIT_PER_PAGE, prepareThreadDisplayData } from "./helper";
+import useEnumOptionsTranslated, { getEnumOptionsForLanguage } from "../../hooks/enums/useEnumOptionsTranslated";
+import { getUsers, getUserInfoExtended, getIsAdmin, getAllOrganizations } from "./service-mail";
+import Pagination from "../pagination/Pagination";
+import InboxMailAdminPanel from "./InboxMailAdminPanel";
+import { UserRole } from "@/submodules/javascript-functions/enums/enums";
+import InboxMailThreadOverview from "./InboxMailThreadOverview";
+import ThreadMailItem from "./InboxMailItem";
+
+
+interface InboxMailViewProps {
+    InboxMailHeader: (props: { children: React.ReactNode }) => JSX.Element;
+    translatorScope: { type: "i18n" | "local"; translator: (key: string) => string };
+    handleInboxMailRefreshToken?: (token: any) => void;
+}
+export default function InboxMailView(props: InboxMailViewProps) {
+
+    const t = props.translatorScope?.translator;
+    const [inboxMailThreads, setInboxMailThreads] = useState<InboxMailThread[]>([]);
+    const [openCreateMail, setOpenCreateMail] = useState(false);
+    const [isNewThread, setIsNewThread] = useState(false);
+    const [currentPage, setCurrentPage] = useState(1);
+    const [fullCount, setFullCount] = useState(0);
+    const [selectedThread, setSelectedThread] = useState<InboxMailThread>(null);
+    const [threadMails, setThreadMails] = useState<InboxMail[]>([]);
+    const [refreshing, setRefreshing] = useState(false);
+    const { state: isAdminSupportThread, setState: setIsAdminSupportThread, ref: isAdminSupportThreadRef } = useRefState(false);
+    const [currentUser, setCurrentUser] = useState(null);
+    const [users, setUsers] = useState<User[]>([]);
+    const [organizations, setOrganizations] = useState([]);
+    const [selectedOrganization, setSelectedOrganization] = useState(null);
+    const [isAdmin, setIsAdmin] = useState<boolean | undefined>(undefined);
+
+    const progressStateOptions = props.translatorScope?.type === "local"
+        ? getEnumOptionsForLanguage(InboxMailThreadSupportProgressState, "InboxMailThreadSupportProgressState", t, "en")
+        : props.translatorScope?.type === "i18n" ? useEnumOptionsTranslated(InboxMailThreadSupportProgressState, "InboxMailThreadSupportProgressState", "enums") : [];
```

Comment:
props not part of change array though probably unproblematic since doesn't change between applications

Reply:
this is not a change array but an empty array